mathics/builtin/numbers/calculus.py

```diff
@@ -812,6 +812,193 @@ def to_sympy(self, expr, **kwargs):
         except Exception:
             return None
 
+ class Reduce(Builtin):
+    
+    """
+    <dl>
+       <dt>'Reduce[$eqs$, $vars$]'
+       <dd> Reduces the statements in $eqs$ by solving equations in $vars$ and eliminating quantifiers.
+    </dl>
+    >> Reduce[x y == 0,{x, y}]
+     = x == 0  ||   y == 0 
+    >> Reduce[{x y == 0, y == 4},{x, y}]
+     = y == 4 && x == 0
+    >> Reduce[{x y == 0, y > 4},{x, y}]
+     =  x == 0 && y > 4
+    >> Reduce[x*y>0,{x,y}]
+     = (x < 0 && y < 0) || (x > 0 && y > 0)
+    >> Reduce[{x*y > 0, y>0},{x,y}]
+     =  x > 0 && y > 0
+    
+    >> Reduce[a<b,a]
+     =  b ∈ Reals && a < b
+    """
+    
+    messages = {
+        "eqf": "`1` is not a well-formed equation.",
+        "svars": 'Equations may not give solutions for all "solve" variables.',
+    }
+
+    rules = {
+        "Reduce[eqs_, vars_, Complexes]": "Reduce[eqs, vars]",
+       
+    }
+
+    summary_text = "find generic solutions for variables, in logical expression form"
+    
+    
+    def apply(self, eqs, vars, evaluation):
+        "Reduce[eqs_, vars_]"
+        
+        vars_original = vars
+        head_name = vars.get_head_name()
+        if head_name == "System`List":
+            vars = vars.leaves
+        else:
+            vars = [vars]
+        for var in vars:
+            if (
+                (var.is_atom() and not var.is_symbol())
+                or head_name in ("System`Plus", "System`Times", "System`Power")  # noqa
+                or constant & var.get_attributes(evaluation.definitions)
+            ):
+
+                evaluation.message("Solve", "ivar", vars_original)
+                return
+        eqs_original = eqs
+        if eqs.get_head_name() in ("System`List", "System`And"):
+            eqs = eqs.leaves
+        else:
+            eqs = [eqs]
+        sympy_eqs = []
+        sympy_denoms = []
+        for eq in eqs:
+            if eq is SymbolTrue:
+                pass
+            elif eq is SymbolFalse:
+                return Expression(SymbolList)
+            elif not eq.has_form("Equal", 2):
+                if any(eq.has_form(operator,2) for operator in ["Less","LessEqual","Greater","GreaterEqual"]):
+                    eq=eq.to_sympy()
+                else:     
+                    return evaluation.message("Solve", "eqf", eqs_original)
+            else:
+                left, right = eq.leaves
+                left = left.to_sympy()
+                right = right.to_sympy()
+                if left is None or right is None:
+                    return
+                eq = left - right
```

Comment:
``eq_sympy = left-rignt``

Reply:
This is not from me.

Reply:
again, this is something that I discovered because you started working on this. It was wrong also in ``Solve``. The idea is not to criticize you, but the code, independently of who wrote it.

Reply:
Done.